### PR TITLE
Converted all usermessages to netmessages

### PR DESCRIPTION
**Pull request by Kefta**
*Phew*

All these changes were rote conversion and are untested. I call out to
all server owners to test these changes and report back any
errors/functionality problems.

Falco, I noticed on FPP_ShareSettings, you left a comment saying you had
issues with the usermessage being too large on servers with >60 people.
I would test the changes you wanted to make with net messages since they
have an increased query size.

Also, I noticed some weird functionality on player hulls. You broadcasted the hull change clientside of a single player, then, set the hull size of that one player on every client. Was this intentional?

Lastly, I left the VOTE:getFilter() function there for legacy purposes, but changed it to return a table instead.

### CODE REVIEW

File: gamemode/modules/voting/sv_votes.lua [hunk 1]
```diff
@@ -50,7 +50,8 @@ function Vote:handleEnd()
 end
 
 function Vote:getFilter()
-	local filter = RecipientFilter()
+	--local filter = RecipientFilter()
```
Comment (Bo98) [inline gamemode/modules/voting/sv_votes.lua]: Unnecessary.

**Comment by Bo98** [comment]
A problematic one could be _Notify. Its hook is currently located in the HUD module which people sometimes disable when they have a custom HUD, and therefor remove the hook into _Notify. Some people fixed this by copy and pasting the hook into their HUDs. This would break it for them. If this change was to happen, then I would recommend moving the hook out of the HUD module:

* It fixes this problem by having the net.Receive in a module that's probably not disabled.
  - It also improves forwards compatibility in case something like this happens in the future again.
* Existing hooks will just be ignored.
  - Potentially could break custom notifications but I'm not even aware of any addon that actually has custom notifications.
  - If any addon were to customise it by hooking the _Notify usermessage then they would be missing Sandbox ones anyway.
* One could argue notifications is more of a GMod thing than a DarkRP HUD thing.

---

As a side note, FPP changes should probably be made in the FPP repo rather than here.

File: gamemode/modules/voting/sv_votes.lua (hunk 1 above)
Comment (Kefta) [inline gamemode/modules/voting/sv_votes.lua]: It's there to show legacy functionality. Eventually, it should just be replaced/removed.

**Comment by Kefta** [comment]
@Bo98 I'm honestly very confused by DarkRP's module -- especially the HUD -- loading. I'll do some research on how to redo it why keeping current functionality, but it will take a bit of time.

**Comment by Bo98** [comment]
Each folder in gamemode/modules is a module. Files are auto loaded from it.

Anyway, my comment was more of a suggestion than what you should definitely do. I'd wait for @FPtje's input.

**Comment by FPtje** [comment]
You're changing fpp code in the DarkRP repository. 
You'll break compatibility with many third party addons, like the before mentioned notifications. 
Many of the changes are pointless. Umsgs still work fine within their limitations, and nothing justifies the risk this commit takes by changing it.
The commit is untested, which is obvious by looking at the code because none of the net messages are registered. 

Sorry, but this is unacceptable. The risk of breaking backwards compatibility is just too high, and it's unnecessary in most cases too. Umsgs should be looked at individually and changed when they're suffering from the umsg limitations.

**Comment by Kefta** [comment]
I'm changing deprecated functions. I didn't register them because I wanted to hear input on the change before I pressed on with messing with the module loading system. I can move all of the FPP code to the appropriate repository.

**Comment by FPtje** [comment]
Umsgs aren't deprecated enough to warrant this risk.

**Comment by Bo98** [comment]
Are you changing other deprecated functions (not umsgs)? I've actually already done that but just haven't committed it yet. Could do that today if it's wanted.

**Comment by Kefta** [comment]
I can make the change Bo suggested to fixing _Notify, however, there will come a point where deprecated functions will actually serve their name and be removed. Willox is already submitting something to Garry's Mod that will create an error upon usage of such a function. The longer you let it fester, the harder it will be for third party developers to update their code to match the main repository.

**Comment by Kefta** [comment]
I am planning on changing everything listed on the Deprecated GMod wiki page.

**Comment by Kefta** [comment]
@FPtje There's no deprecation amount: it's either deprecated or not, and people will start getting errors for it when Robotboy finds a proper solution in a refreshing list for deprecation.

**Comment by Bo98** [comment]
> I am planning on changing everything listed on the Deprecated GMod wiki page. There have been new additions in the past week to it.

Yeah I did this last week, but wanted to test it more before committing.

**Comment by FPtje** [comment]
Notify isn't the only thing. Potentially every usermessage is listened to by a third party addon. **Many** addons will break by this change and therefore I absolutely refuse it until it's absolutely necessary to change. Otherwise I will get the blame for arbitrarily breaking so many addons, and rightly so. 

I'll change it when warnings start popping up.

**Comment by Kefta** [comment]
@FPtje I'll personally assure collateral on fixing any addons that are broken by this update

**Comment by FPtje** [comment]
You cannot promise the impossible. I'd like to see you hack into scriptfodder to update the addons on there.

**Comment by aStonedPenguin** [comment]
you all care way too much. let the children cry.

**Comment by FPtje** [comment]
Not an option.

**Comment by Kefta** [comment]
If there are problems with authors not maintaining their addons, I will fix anyone's copy who is having problems.

**Comment by FPtje** [comment]
For fuck's sake, batman. You cannot fix every addon that every fucking server owner is running. You cannot fix the shitstorm of addons breaking by your change and I **definitely** cannot trust you to do it.

**Comment by Kefta** [comment]
So never again are improvements that break a few eggshells allowed? We are stuck always catering to a third-party? I fully understand your concern, but as we learned from Garry's Mod 13, pushing things off is a catastrophe waiting to occur. You should make a change as early as possible if the longevity is beneficial: this should have been done already.

**Comment by aStonedPenguin** [comment]
According to this informative wikihow article crying actually makes you feel better right away!
http://www.wikihow.com/Cry-and-Let-It-All-Out

**Comment by thelastpenguin** [comment]
Send both a user message and a net message for backwards compatability.

It's the best of both worlds I swear.

People can stay autistic.
Something gets fixed.
FPP becomes even more of a performance issue than before.

Little actually changes.

**Comment by FPtje** [comment]
DarkRP has a large API. An API is a responsibility and a promise that it will work. Breaking that promise rightfully hurts the trust in the API. 

Backwards compatibility should not be broken unless absolutely necessary. Developers get pissed when every update ever always breaks compatibility. You see that with gmod updates. It pisses people off, and rightfully so. 

I mean why *shouldn't* they be pissed off? They made something, put a lot of effort in it and found that it **worked** for ages until *someone else* came along and decided that it shouldn't work anymore. Now *they* have to go back to the addon to fix it. This often doesn't happen, pissing off the people using the unmaintained software.

It's really fucking simple. Breaking backwards compatibility pisses people off, and rightfully so.

**Comment by RonPaul2012** [comment]
Yes but falco as stated in the article crying will make them feel better. It solves itself basically. ya know!?

**Comment by thelastpenguin** [comment]
Ik. That's what I'm saying.

Keep both. It's backwards compatible and, as everyone knows, two is better than one.

**Comment by FPtje** [comment]
Fuck off

**Comment by RonPaul2012** [comment]
i mean in all honesty i just need darkrp to work on my new server cus like srcds doesnt even fukin strart on my windows 2003 idk wtf to do

**Comment by thelastpenguin** [comment]
@FPtje In seriousness Id like to make an actual argument in favor of changing this. People simply should not be writing code that depends on internal implementations, their code should be based on exposed methods. They are doing something wrong if they are doing anything else. It might be agood idea to create a pull request that exposes these methods if they are not exposed in any other manor, but simply because we might break a dependency on an internal implementation is not a good arguemnt for not making a change to fix an actual issue that affects some servers.

Honestly I'd say fix the issue and let the addons break simply because they should not be doing waht they are doing. Yes they might be annoyed but that's not your problem Your problem is ensuring that your actually exposed API and not the internal implementation conforms to what people expect from it. Your priority here should be ensuring that darkrp itself works for all of it's users, it is upto the developers of those addons to ensure that their addons work for their users.

**Comment by Kefta** [comment]
But as all developers know, changes are made that break things in the sake of efficiency and cleanliness. It's a necessary evil in the world of programming.

**Comment by PHPApple** [comment]
>DarkRP has a large API. An API is a responsibility and a promise that it will work. Breaking that promise rightfully hurts the trust in the API.

Read this in a stereotypical republican southern baptist voice. Falco sounds like motherfucking George Busch running a country.

**Comment by RonPaul2012** [comment]
Yeah I mean I think code is right falco, catering to a majority of a playerbase that purchases scripts off of a third party website to use in this autistic cluster fucks they call "servers" isn't the coolest thing to do right? I mean, what ever happend to encouraging development or uniqueness? Nope! I gotta go buy that new sweet MOTD on scriptfodder because I'm 16 and don't want to put any effort into being unique. WOO THIRD PARTY SCRIPTS! YA SHITTY GMOD SERVERS! LET'S CATER TO THE PROBLEM! WOO!

**Comment by FPtje** [comment]
The efficiency gain in changing a constant length usermessage to a similar net message is negligible.

I'm not necessarily saying "*don't fix it when it's not broken*", I'm saying "*Fixing it will break more than it's worth*". **Until changing this becomes worth it, it is not worth it**. The efficiency gain is absolutely negligible and no third party developer nor end user cares about cleanliness. Changing this will piss developers off, and then telling them that "*they shouldn't have been relying on internal umsgs in the first place*" will piss them off even more, especially since there were(/are?) no alternatives available. That would be screwing them over and then blaming them for it.

The reasoning is **very** simple. This change will fuck things up and I will not do it unless it's **necessary**. I'll be happy to do it when it **does** become necessary.

Also, @RonPaul2012 you contribute nothing but incompetence to this discussion.

**Comment by Kefta** [comment]
Better yet: we all concede in not merging this immediately, but warn server owners and developers now of the change with plenty of time. This way, active developers can fix their scripts and old developers can test their current addons on a sort of "beta" gamemode. It's fair, and keeps trust in the API.

**Comment by RonPaul2012** [comment]
im 13 and gay

**Comment by RonPaul2012** [comment]
help dad

**Comment by aStonedPenguin** [comment]
> no third party developer nor end user cares about cleanliness
I don't even have words for that.

**Comment by PHPApple** [comment]
>Yeah I mean I think code is right falco, catering to a majority of a playerbase that purchases scripts off of a third party website to use in this autistic cluster fucks they call "servers" isn't the coolest thing to do right? I mean, what ever happend to encouraging development or uniqueness? Nope! I gotta go buy that new sweet MOTD on scriptfodder because I'm 16 and don't want to put any effort into being unique. WOO THIRD PARTY SCRIPTS! YA SHITTY GMOD SERVERS! LET'S CATER TO THE PROBLEM! WOO!

You don't understand, falco is actually the biological father to all children of gmod and scriptfodder. He has to take care of them and make sure everything works.

**Comment by FPtje** [comment]
Alright, that's the end of @RonPaul2012

@Kefta Your idea is the way to go, but only if umsgs will indeed give trouble in a future update.

Also the end of @PHPApple.

**Comment by SenorOllie** [comment]
Hi

**Comment by Kefta** [comment]
https://github.com/garrynewman/garrysmod/pull/826

Robotboy already said he will be merging this eventually after Kilburn makes some changes to the erroring system. I will also tell Willox to update the PR to the wiki page.

**Comment by Pengu1nZ** [comment]
While you do have some interesting points, I would argue Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras erat diam, molestie a sollicitudin cursus, condimentum non est. Etiam sit amet purus tincidunt, auctor massa at, luctus leo. Maecenas tempor mauris et ipsum faucibus cursus. Donec diam neque, pharetra a facilisis et, placerat dictum mi. Suspendisse fermentum sollicitudin blandit. Phasellus pharetra vehicula lacus, id pretium sem semper vitae. Sed vestibulum laoreet neque, ac tincidunt nunc euismod et.

Mauris faucibus mollis magna quis eleifend. Nullam ut dui at ex blandit ullamcorper vitae vitae tortor. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Donec id laoreet erat. Cras rhoncus, eros vitae venenatis tempor, nisi tortor vestibulum ex, rhoncus placerat neque augue id purus. Etiam pulvinar sem non augue interdum, non imperdiet arcu porta. Donec imperdiet ultrices fringilla. Donec pellentesque, felis ut tempor finibus, diam mauris suscipit lectus, eu tempor nisi orci id diam.

Quisque ut ipsum elementum mi tristique efficitur vitae sed massa. Cras sed ex imperdiet, commodo eros et, tempus eros. Mauris lacus justo, accumsan eu consequat et, convallis id ipsum. Suspendisse tempor consectetur augue feugiat ultrices. Nullam dui mauris, auctor id vestibulum quis, pellentesque a augue. Pellentesque vitae lacinia est. Phasellus luctus leo ut mi varius congue. Integer vitae sollicitudin ipsum, vitae maximus ipsum. Sed fermentum erat sed tellus placerat, in tristique mauris vulputate. Praesent pretium faucibus sapien eget sodales.

Vestibulum faucibus, mi eu ullamcorper ultrices, nisl velit sollicitudin eros, sed rutrum lectus neque eget leo. Sed vel pharetra tortor. Donec posuere ipsum vitae ante feugiat, eu faucibus nunc vulputate. Sed volutpat metus id odio mollis, a bibendum nunc congue. Integer auctor diam eu suscipit placerat. Aenean tristique nibh eget enim lobortis, eget convallis turpis vehicula. Suspendisse porta iaculis volutpat. Quisque tempus vestibulum mauris eget imperdiet. Cras cursus nunc a rutrum feugiat.

Fusce sed lacus volutpat velit semper aliquet. Nullam ac ultrices nisl. Aliquam quis ante vitae tortor euismod pulvinar. Morbi volutpat pharetra nisl nec vulputate. Donec efficitur volutpat diam, a scelerisque purus suscipit quis. Ut hendrerit diam sed sodales dictum. Aliquam finibus condimentum ligula, vel facilisis enim feugiat vitae. U. That is all my say on this matter

**Comment by FPtje** [comment]
usermessages aren't deprecated in that PR.

**Comment by Kefta** [comment]
They weren't on the list at the time the PR was made, however, they are deprecated on the wiki. I will ask Willox to update the PR.

**Comment by HDG-MrDeeJay** [comment]
Whats with the army of trolls? Or is it just two guys creating accounts....1

**Comment by FPtje** [comment]
So this whole pull request is here because you are going to *ask* willox to put umsgs on the deprecation list?

That's the second time in a week or so that someone makes a pull request for DarkRP based on their opinion of something that should change in GMod. What the fuck, people.

**Comment by Kefta** [comment]
It's already deprecated. If you read Willox's commit description, you would know that deprecation page is the universal guide to what is being added. He's not going to add 5000 commits for every function that's going to be deprecated over time. I'm only asking him to confirm for you that it's going to cause errors so that it will satisfy your desire for evidence.

**Comment by Kefta** [comment]
Willox has come up with alternative to redirecting all usermessages in C++ to net instead, thus you can send a message with net and receive with um if you wanted

**Comment by FPtje** [comment]
Yup, I talked to him about it. Changing everything now would indeed be the
worst option.

**Pull request closed without merging**